### PR TITLE
storage: Implement StorageHTTPSigner test

### PR DESCRIPTION
**Pull request by JinnyYi**
ref: https://github.com/beyondstorage/go-storage/issues/707

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/beyondstorage/go-integration-test/v4
 go 1.15
 
 require (
-	github.com/beyondstorage/go-storage/v4 v4.4.0
+	github.com/beyondstorage/go-storage/v4 v4.4.1-0.20210819041350-cf8c0edf961c
```
Comment (Xuanwo) [inline go.mod]: Maybe we need a `v4.5.0`.

File: http_signer.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+		Convey("When write via QuerySignHTTP", func() {
+			size := rand.Int63n(4 * 1024 * 1024)
+			content, err := ioutil.ReadAll(io.LimitReader(randbytes.NewRand(), size))
+			if err != nil {
+				t.Error(err)
+			}
+
+			path := uuid.New().String()
+			req, err := signer.QuerySignHTTP(types.OpStoragerWrite, path, time.Duration(time.Hour))
```
Comment (Xuanwo) [inline http_signer.go]: Do we need to set `size` here?

I think we need a demo to make sure our test logic is correct. Maybe in s3?

Comment (JinnyYi) [reply http_signer.go]: > Do we need to set `size` here?

Dose it mean set `Content-Length`? I tried to add `req.Header.Add("Content-Length", strconv.FormatInt(size,10))` and tested with qingstor, the request was successful, but the request is chunked transfer encoding. So I removed the content-length setting.

Comment (JinnyYi) [reply http_signer.go]: According to the comment for `Header` in `http.Request`, for client requests, certain headers such as Content-Length and Connection are automatically written when needed and values in Header may be ignored.

Comment (JinnyYi) [reply http_signer.go]: > I think we need a demo to make sure our test logic is correct. Maybe in s3?

I'll privde a demo.

Comment (Xuanwo) [reply http_signer.go]: > chunked transfer encoding

Strange, chunked transfer encoding should be used for requests that don't have `content-length` header.

Let's discuss the detail in the demo of s3 instead.

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**